pytorch_lightning/trainer/connectors/data_connector.py

```diff
@@ -250,6 +245,40 @@ def detach_data(model: "pl.LightningModule") -> None:
             if isinstance(loader, _PatchDataLoader):
                 loader.unpatch(model)
 
+    def teardown(self) -> None:
+        items = list(vars(self).items())
+        for attr_name, attr in items:
+            if isinstance(attr, AbstractDataFetcher):
+                attr.teardown()
+                delattr(self, attr_name)
+
+    def _check_training_step_requires_dataloader_iter(self) -> bool:
```

Comment:
don't we want to add this to ConfigValidator?

Reply:
Done !